daemon/kill.go

```diff
@@ -162,9 +161,6 @@ func (daemon *Daemon) Kill(container *containerpkg.Container) error {
 	log.G(ctx).WithError(status.Err()).WithField("container", container.ID).Errorf("Container failed to exit within %v of kill - trying direct SIGKILL", waitTimeout)
 
 	if err := killProcessDirectly(container); err != nil {
-		if errors.As(err, &errNoSuchProcess{}) {
-			return nil
-		}
 		return err
 	}
```

Comment:
Hm... right, so I _think_ I understand the problem. Previously we were returning early on an _ERROR_  but only if that error was `errNoSuchProcess`. But if there was _NO_ error, we would continue the code.

So I think there's a change in logic in my first 2 commits, which could either be "correct" (but code doesn't handle it well), or incorrect. Need to do some more digging

I'll move the other two commits to a separate PR